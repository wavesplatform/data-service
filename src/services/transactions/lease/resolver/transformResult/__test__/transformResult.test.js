const transformResultsOne = require('../one');
const transformResultsMany = require('../many');
const txRaw = require('./mocks/tx');
const Maybe = require('folktale/maybe');

const request = {
  sort: 'asc',
};
describe('Transfer transactions transformResult function', () => {
  it('works for one', () => {
    expect(transformResultsOne(Maybe.Just(txRaw))).toMatchSnapshot();
  });
  it('works for many', () => {
    expect(
      transformResultsMany(
        [
          Maybe.Just(txRaw),
          Maybe.Just(txRaw),
          Maybe.Nothing(),
          Maybe.Just(txRaw),
        ],
        request
      )
    ).toMatchSnapshot();
  });
});
