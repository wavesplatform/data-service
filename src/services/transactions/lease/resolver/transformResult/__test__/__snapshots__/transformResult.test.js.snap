// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transfer transactions transformResult function works for many 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "10705754895",
        "fee": "100000",
        "height": 600887,
        "id": "2rdiawoYXnwVieRgTkESYKAKwcRrnLZQ4abmC9JQSj7n",
        "recipient": "3P2HNUd5VUPLMQkJmctTPEeeHumiPN2GkTb",
        "sender": "3PMY5XbSpvcQnYNCMiLWAaHuTwQwN6sSoy4",
        "senderPublicKey": "AiXERNpnVgGM25Au55kfcz39K9FUoysqD5nCSpew4MYo",
        "signature": "5CVsdA6Weyx28MyGAQoFD8HdSBMmop3RQJvGv8w4k2axRyJ4f76oV4vWoCoV31B4Dv2dRcSfG6N88AzszccH9xV",
        "timestamp": 2017-07-29T02:22:01.407Z,
        "type": 8,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "10705754895",
        "fee": "100000",
        "height": 600887,
        "id": "2rdiawoYXnwVieRgTkESYKAKwcRrnLZQ4abmC9JQSj7n",
        "recipient": "3P2HNUd5VUPLMQkJmctTPEeeHumiPN2GkTb",
        "sender": "3PMY5XbSpvcQnYNCMiLWAaHuTwQwN6sSoy4",
        "senderPublicKey": "AiXERNpnVgGM25Au55kfcz39K9FUoysqD5nCSpew4MYo",
        "signature": "5CVsdA6Weyx28MyGAQoFD8HdSBMmop3RQJvGv8w4k2axRyJ4f76oV4vWoCoV31B4Dv2dRcSfG6N88AzszccH9xV",
        "timestamp": 2017-07-29T02:22:01.407Z,
        "type": 8,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": null,
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "10705754895",
        "fee": "100000",
        "height": 600887,
        "id": "2rdiawoYXnwVieRgTkESYKAKwcRrnLZQ4abmC9JQSj7n",
        "recipient": "3P2HNUd5VUPLMQkJmctTPEeeHumiPN2GkTb",
        "sender": "3PMY5XbSpvcQnYNCMiLWAaHuTwQwN6sSoy4",
        "senderPublicKey": "AiXERNpnVgGM25Au55kfcz39K9FUoysqD5nCSpew4MYo",
        "signature": "5CVsdA6Weyx28MyGAQoFD8HdSBMmop3RQJvGv8w4k2axRyJ4f76oV4vWoCoV31B4Dv2dRcSfG6N88AzszccH9xV",
        "timestamp": 2017-07-29T02:22:01.407Z,
        "type": 8,
        "version": 1,
      },
    },
  ],
  "lastCursor": "MjAxNy0wNy0yOVQwMjoyMjowMS40MDdaOjoycmRpYXdvWVhud1ZpZVJnVGtFU1lLQUt3Y1JybkxaUTRhYm1DOUpRU2o3bjo6YXNj",
}
`;

exports[`Transfer transactions transformResult function works for one 1`] = `
Object {
  "__type": "transaction",
  "data": Object {
    "amount": "10705754895",
    "fee": "100000",
    "height": 600887,
    "id": "2rdiawoYXnwVieRgTkESYKAKwcRrnLZQ4abmC9JQSj7n",
    "recipient": "3P2HNUd5VUPLMQkJmctTPEeeHumiPN2GkTb",
    "sender": "3PMY5XbSpvcQnYNCMiLWAaHuTwQwN6sSoy4",
    "senderPublicKey": "AiXERNpnVgGM25Au55kfcz39K9FUoysqD5nCSpew4MYo",
    "signature": "5CVsdA6Weyx28MyGAQoFD8HdSBMmop3RQJvGv8w4k2axRyJ4f76oV4vWoCoV31B4Dv2dRcSfG6N88AzszccH9xV",
    "timestamp": 2017-07-29T02:22:01.407Z,
    "type": 8,
    "version": 1,
  },
}
`;
